Clippy warns about `drop` on `struct` with `#[wasm_bindgen]` attribute that also derives `Copy`

### Describe the Bug

I want to embed a struct inside another struct and both structs have the `#[wasm_bindgen]` attribute:

If I just do it the naive way:

```rust
#[wasm_bindgen(js_name = Outer)]
pub struct Outer {
    #[wasm_bindgen]
    pub inner: Option<Inner>,
}

#[wasm_bindgen(js_name = Inner)]
pub struct Inner {
    #[wasm_bindgen]
    pub v: f64,
}
```

I get a compile error:


```shell
error[E0277]: the trait bound `inner::Inner: std::marker::Copy` is not satisfied
  --> src/outer.rs:11:16
   |
11 |     pub inner: Option<Inner>,
   |                ^^^^^^^^^^^^^ the trait `std::marker::Copy` is not implemented for `inner::Inner`
   |
   = note: required because of the requirements on the impl of `std::marker::Copy` for `std::option::Option<inner::Inner>`
note: required by a bound in `outer::__wbg_get_outer_inner__const::__wbg_get_outer_inner::assert_copy`
  --> src/outer.rs:8:1
   |
8  | #[wasm_bindgen(js_name = Outer)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `outer::__wbg_get_outer_inner__const::__wbg_get_outer_inner::assert_copy`
   = note: this error originates in the attribute macro `wasm_bindgen` (in Nightly builds, run with -Z macro-backtrace for more info)
```

So I do what the compiler tells me, I make `Inner` derive `Clone` and `Copy`:

```rust
#[wasm_bindgen(js_name = Inner)]
#[derive(Clone, Copy)]
pub struct Inner {
    #[wasm_bindgen]
    pub v: f64,
}
```

Now the compiler is happy 🤗 but clippy is sad 😢:

```shell
error: calls to `std::mem::drop` with a value that implements `Copy`. Dropping a copy leaves the original intact
 --> src/inner.rs:7:1
  |
7 | #[wasm_bindgen(js_name = Inner)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[deny(clippy::drop_copy)]` on by default
note: argument has type inner::Inner
 --> src/inner.rs:7:1
  |
7 | #[wasm_bindgen(js_name = Inner)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#drop_copy
  = note: this error originates in the attribute macro `wasm_bindgen` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Now I can just silence 🤫 Clippy, but I checked issues on this repository and no one seems to have reported this "bug" before, so I thought maybe it would help the project.

I did a `cargo expand` on `Inner` and the only reference to `drop` is here:

```
    #[cfg(all(target_arch = "wasm32", not(target_os = "emscripten")))]
    #[automatically_derived]
    const __wbg_inner_free__const: () = {
        #[no_mangle]
        #[doc(hidden)]
        pub unsafe extern "C" fn __wbg_inner_free(ptr: u32) {
            drop(<Inner as wasm_bindgen::convert::FromWasmAbi>::from_abi(ptr));
        }
        ()
    };
```

### Steps to Reproduce

1. Implement a struct hierarchy using `wasm_bindgen`
2. Make the inner struct `Clone` and `Copy`
3. Run `cargo clippy`

### Expected Behavior

Maybe `wasm_bindgen` could skip the drop and make Clippy happy 😘?

### Actual Behavior

Clippy is sad 😢

--- a/crates/backend/src/codegen.rs
+++ b/crates/backend/src/codegen.rs
@@ -225,7 +225,7 @@ impl ToTokens for ast::Struct {
                 #[no_mangle]
                 #[doc(hidden)]
                 pub unsafe extern "C" fn #free_fn(ptr: u32) {
-                    drop(<#name as wasm_bindgen::convert::FromWasmAbi>::from_abi(ptr));
+                    let _ = <#name as wasm_bindgen::convert::FromWasmAbi>::from_abi(ptr); //implicit `drop()`
                 }
             };
 

